chain/deals/client_storagemarket.go

```diff
@@ -0,0 +1,134 @@
+package deals
+
+// this file implements storagemarket.StorageClient
+
+import (
+	"context"
+
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/storagemarket"
+)
+
+func (c *Client) ListProviders(ctx context.Context) (<-chan storagemarket.StorageProviderInfo, error) {
+	providers, err := c.node.ListStorageProviders(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make(chan storagemarket.StorageProviderInfo)
+
+	go func() {
+		for _, p := range providers {
+			select {
+			case out <- *p:
+			case <-ctx.Done():
+				return
+			}
+
+		}
+	}()
+
+	return out, nil
+}
+
+func (c *Client) ListDeals(ctx context.Context) ([]actors.OnChainDeal, error) {
+	addr, err := c.node.GetDefaultWalletAddress(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.node.ListClientDeals(ctx, addr)
+}
+
+func (c *Client) ListInProgressDeals(ctx context.Context) ([]storagemarket.ClientDeal, error) {
+	deals, err := c.List()
+	if err != nil {
+		return nil, err
+	}
+
+	out := make([]storagemarket.ClientDeal, len(deals))
+	for k, v := range deals {
+		out[k] = storagemarket.ClientDeal{
+			ProposalCid:    v.ProposalCid,
+			Proposal:       v.Proposal,
+			State:          v.State,
+			Miner:          v.Miner,
+			MinerWorker:    v.MinerWorker,
+			DealID:         v.DealID,
+			PublishMessage: v.PublishMessage,
+		}
+	}
+
+	return out, nil
+}
+
+func (c *Client) GetInProgressDeal(ctx context.Context, cid cid.Cid) (storagemarket.ClientDeal, error) {
+	deals, err := c.ListInProgressDeals(ctx)
+	if err != nil {
+		return storagemarket.ClientDeal{}, err
+	}
+
+	for _, deal := range deals {
+		if deal.ProposalCid == cid {
+			return deal, nil
+		}
+	}
+
+	return storagemarket.ClientDeal{}, xerrors.Errorf("couldn't find client deal")
+}
+
+func (c *Client) GetAsk(ctx context.Context, info storagemarket.StorageProviderInfo) (*storagemarket.StorageAsk, error) {
+	return c.QueryAsk(ctx, info.PeerID, info.Address)
```

Comment:
to insure the old functions aren't getting called, I might consider making QueryAsk, Start, etc private